packages/hydrogen-react/src/useMoney.tsx

```diff
@@ -115,38 +115,41 @@ export function useMoney(money: MoneyV2): UseMoneyValue {
 
   const amount = parseFloat(money.amount);
 
-  const options = useMemo(
-    () => ({
+  const {
+    defaultFormatter,
+    nameFormatter,
+    narrowSymbolFormatter,
+    withoutTrailingZerosFormatter,
+    withoutCurrencyFormatter,
+    withoutTrailingZerosOrCurrencyFormatter,
+  } = useMemo(() => {
+    const options = {
       style: 'currency' as const,
       currency: money.currencyCode,
-    }),
-    [money.currencyCode],
-  );
-
-  const defaultFormatter = useLazyFormatter(locale, options);
-
-  const nameFormatter = useLazyFormatter(locale, {
-    ...options,
-    currencyDisplay: 'name',
-  });
-
-  const narrowSymbolFormatter = useLazyFormatter(locale, {
-    ...options,
-    currencyDisplay: 'narrowSymbol',
-  });
-
-  const withoutTrailingZerosFormatter = useLazyFormatter(locale, {
-    ...options,
-    minimumFractionDigits: 0,
-    maximumFractionDigits: 0,
-  });
```

Comment:
Instead of calling `useLazyFormatter` many times, each with a new object in memory for each render, put it all in one memo above. This makes `useLazyFormatter` calls much fewer.